gradoop-data-integration/src/main/java/org/gradoop/dataintegration/transformation/EdgeToVertex.java

```diff
@@ -0,0 +1,206 @@
+   *
+   * @param edgeLabel            The label of the edges use for the transformation.
+   *                             (No edges will be transformed if this parameter is {@code null}).
+   * @param newVertexLabel       The label of the newly created vertex.
+   * @param edgeLabelSourceToNew The label of the newly created edge which points to the newly
+   *                             created vertex.
+   * @param edgeLabelNewToTarget The label of the newly created edge which starts at the newly
+   *                             created vertex.
+   *
+   * @throws NullPointerException when at least on of the latter 3 arguments is {@code null}.
+   */
+  public EdgeToVertex(String edgeLabel, String newVertexLabel, String edgeLabelSourceToNew,
+    String edgeLabelNewToTarget) {
+    Objects.requireNonNull(newVertexLabel);
+    Objects.requireNonNull(edgeLabelSourceToNew);
+    Objects.requireNonNull(edgeLabelNewToTarget);
+    this.edgeLabel = edgeLabel;
+    this.newVertexLabel = newVertexLabel;
+    this.edgeLabelSourceToNew = edgeLabelSourceToNew;
+    this.edgeLabelNewToTarget = edgeLabelNewToTarget;
+  }
+
+  @Override
+  public LogicalGraph execute(LogicalGraph graph) {
+    DataSet<Edge> relevantEdges = graph.getEdgesByLabel(edgeLabel);
+
+    // create new vertices
+    DataSet<Tuple3<Vertex, GradoopId, GradoopId>> newVerticesAndOriginIds =
+        relevantEdges
+            .map(new CreateVertexFromEdges(newVertexLabel, graph.getConfig().getVertexFactory()));
+
+    DataSet<Vertex> newVertices =
+        newVerticesAndOriginIds
+            .map(new Value0Of3<>())
+            .union(graph.getVertices());
+
+    // create edges to the newly created vertex
+    DataSet<Edge> newEdges =
+        newVerticesAndOriginIds
+            .flatMap(new CreateEdgesFromTriple(graph.getConfig().getEdgeFactory(),
+              edgeLabelSourceToNew, edgeLabelNewToTarget))
+            .union(graph.getEdges());
+
+    return graph.getConfig().getLogicalGraphFactory()
+      .fromDataSets(newVertices, newEdges);
+  }
+
+  @Override
+  public String getName() {
+    return this.getClass().getName();
+  }
+
+  /**
+   * A {@link MapFunction} that creates a new vertex based on the given edge. Furthermore it
+   * returns the source and target id of the edge for later use.
+   */
+  private static class CreateVertexFromEdges implements MapFunction<Edge, Tuple3<Vertex, GradoopId,
+    GradoopId>> {
+
+    /**
+     * The factory vertices are created with.
+     */
+    private final VertexFactory factory;
+
+    /**
+     * The label of the newly created vertex.
+     */
+    private final String newVertexLabel;
+
+    /**
+     * The constructor of the MapFunction.
+     *
+     * @param newVertexLabel The label of the newly created vertex.
+     * @param factory        The factory for creating new vertices.
+     */
+    CreateVertexFromEdges(String newVertexLabel, VertexFactory factory) {
+      this.factory = factory;
```

Comment:
Same here. Instead of holding the factory, create a vertex instance from the factory that can be reused